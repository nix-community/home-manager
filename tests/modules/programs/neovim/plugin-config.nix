{ config, lib, pkgs, realPkgs, ... }:

lib.mkIf config.test.enableBig {
  programs.neovim = {
    enable = true;
    extraConfig = ''
      let g:hmExtraConfig='HM_EXTRA_CONFIG'
    '';
    plugins = with pkgs.vimPlugins; [
      vim-nix
      {
        plugin = vim-commentary;
        config = ''
          let g:hmPlugins='HM_PLUGINS_CONFIG'
        '';
      }
    ];
    extraLuaPackages = [ pkgs.lua51Packages.luautf8 ];
  };

  _module.args.pkgs = lib.mkForce realPkgs;

  nmt.script = ''
    vimout=$(mktemp)
    echo "redir >> /dev/stdout | echo g:hmExtraConfig | echo g:hmPlugins | redir END" \
      | ${pkgs.neovim}/bin/nvim -es -u "$TESTED/home-files/.config/nvim/init.lua" \
      > "$vimout" || true
    assertFileContains "$vimout" "HM_EXTRA_CONFIG"
    assertFileContains "$vimout" "HM_PLUGINS_CONFIG"
  '';
}

