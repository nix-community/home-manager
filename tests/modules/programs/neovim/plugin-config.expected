vim.cmd [[source /nix/store/00000000000000000000000000000000-nvim-init-home-manager.vim]]
function HM_PLUGIN_LUA_CONFIG ()
end
